### ISSUE TITLE
Add catalyse method to RestrictionBatch

### ISSUE THREAD

**Issue by chenzhiw** [biopython/biopython#4285]
[RestrictionBatch does not implement a catalyse method, as it would not have a real meaning when used with large batch.](https://github.com/biopython/biopython/blob/master/Doc/cookbook/Restriction/Restriction.md#2.6)

It dose have meaning!!! We usually use two enzymes to digest a plasmid.
Is it possible to add catalyse method to RestrictionBatch.

**Comment by MarkusPiotrowski** [comment]
@chenzhiw The statement is _when used with large batches_ which is for sure correct. But even when doing a simple double digest (a typical situation in cloning) I wonder what would be your use case, so why do you want to do this, and what is your expected result? This will help us to think more about this.

E.g. you do a digest with EcoRI as shown in the tutorial:
```python
>>> EcoRI.catalyze(ecoseq)
(Seq('AAAAAAAAAAAAAAG'), Seq('AATTCAAAAAAAAAAAAAA'))
```
However, the real situation looks different, you get two partially double-stranded DNA molecules with overhangs:
```
AAAAAAAAAAAAAAG       +   AATTCAAAAAAAAAAAAAA
TTTTTTTTTTTTTTCTTAA           GTTTTTTTTTTTTTT
``` 

Now, suggest your second enzyme would cut in AATT. So would this cut these fragments at their borders (I don't think so, because they are single stranded)? 

So if you just want to know the (approx. !) size of some fragments in a double or n digest, a `catalyze` method with a `RestrictionBatch` could make sense, but if you have something in mind which is closer to 'virtual cloning` this wouldn't help, because the resulting sequences are missing the crucial information about overhangs.

**Comment by chenzhiw** [comment]
@MarkusPiotrowski Thanks for your reply. I understand your concern. Multiple enzymes will make the problem complex.
My problem is if I digest the [plasmid pUC57](https://www.snapgene.com/plasmids/basic_cloning_vectors/pUC57) with EcoRI and ScaI, what are the size of fragmens. Similar to the [Simulate Agarose Gel tool of snapgene](https://support.snapgene.com/hc/en-us/articles/10384180657684-Create-an-Agarose-Gel).

Is it possible if we do not consider the complex condition like overlaped sites etc.?

**Comment by MarkusPiotrowski** [comment]
Maybe something like `.fragments` could be a useful supplement to the `Analysis` class.
For now, you may consider this (it isn't possibly the most elegant solution, but it seems to work):
```python
from Bio import SeqIO
from Bio import Restriction
from Bio.Restriction import Analysis, RestrictionBatch

# Read the sequence and find out if it is linear
my_record = SeqIO.read('pUC57.dna', 'snapgene')
linear = False if my_record.annotations["topology"] == 'circular' else True
my_seq = my_record.seq

# Do the restriction analysis
my_enzyme_batch = RestrictionBatch([Restriction.EcoRI, Restriction.ScaI])
my_analysis = Analysis(my_enzyme_batch, my_seq, linear=linear)
analysis_result = my_analysis.full()

# From the results, get all the cutting positions as flattened list
cutting_sites = [val for key, value in analysis_result.items() for val in value]

# Add start and end positions (linear) or 'virtual end position' (circular)
if linear:
    cutting_sites += [0, len(my_seq)]
else:
    cutting_sites += [len(my_seq) + min(cutting_sites)]

# Now calculate the fragments' sizes as consecutive deltas of the list entries
fragments = []
for n, site in enumerate(sorted(cutting_sites)):
    if n > 0:
        fragments.append(site - before_site)
    before_site = site

print(sorted(fragments))
```
Result: `[903, 1807]`

**Comment by chenzhiw** [comment]
Yeah, thx, this is indeed a workable solution. I use RestrictionBatch.search function with the same affect.

**Comment by manulera** [comment]
I can recommend https://github.com/BjornFJohansson/pydna for this use-case. Has subclasses of Biopython sequences representing double strand sequences.
